### PR TITLE
Dev 3946 array selector

### PR DESCRIPTION
**Pull request by nprimo**

### CODE REVIEW

File: subjects/devops/array-selector/README.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+## array-selector
+
+### Instructions
+
+Create a script `array-selector.sh`, which will have an array declared with the
+following values (in this order): `red`, `blue`, `green`, `white`, `black`.
+
+When executed, the script will try to print the element in the position specified in the first argument passed to the script.
+
+The script should return `Error` when:
```
Comment (MSilva95) [inline subjects/devops/array-selector/README.md:10]: Just an opinion:
```
The script should return `Error` when:

- the number of given arguments is different from one,

- the given argument is not a number,

- the given argument is a number outside the range of the array.
```

File: subjects/devops/array-selector/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
+## array-selector
+
+### Instructions
+
+Create a script `array-selector.sh`, which will have an array declared with the
+following values (in this order): `red`, `blue`, `green`, `white`, `black`.
+
+When executed, the script will try to print the element in the position specified in the first argument passed to the script.
```
Comment (MSilva95) [inline subjects/devops/array-selector/README.md:8]: the element at the position

File: sh/tests/array-selector_test.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
```
Comment (MSilva95) [inline sh/tests/array-selector_test.sh:1]: this file is not executable.
Also the test passes with an empty solution

File: sh/tests/solutions/array-selector.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
```
Comment (MSilva95) [inline sh/tests/solutions/array-selector.sh:1]: this file is not executable

File: sh/tests/array-selector_test.sh [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+challenge() {
+    submitted=$(bash "$script_dirS"/student/array-selector.sh $1)
+    expected=$(bash "$script_dirS"/solutions/array-selector.sh $1)
+
+    diff <(echo "$submitted") <(echo "$expected")
+}
+
+# Test with numbers - out of range included
+for num in {0..6}
```
Comment (MSilva95) [inline sh/tests/array-selector_test.sh:19]: the for is not working i thin it is because it starts at 0 and 0 is supposed to give you error and it is not working 
If you do the for like so:
```
for num in {1..6}; do
    challenge $num
done
```
it will be a valid test for the correct values of the array, but it will still fail for the error part

File: subjects/devops/array-selector/README.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
+## array-selector
+
+### Instructions
+
+Create a script `array-selector.sh`, which will have an array declared with the
```
Comment (MSilva95) [inline subjects/devops/array-selector/README.md:5]: Explain somewhere that this array will work different than normal as it will start at position 1 instead of 0.

File: subjects/devops/array-selector/README.md (hunk 1 above)
Comment (nprimo) [reply subjects/devops/array-selector/README.md:10]: I like it better, I will change it like you suggested :)

File: sh/tests/solutions/array-selector.sh (hunk 4 above)
Comment (nprimo) [reply sh/tests/solutions/array-selector.sh:1]: I will change this just to avoid weird behaviour, but it should not be a problem since the test script is running `bash <filename>`

File: subjects/devops/array-selector/README.md (hunk 6 above)
Comment (nprimo) [reply subjects/devops/array-selector/README.md:5]: I have added a sentence to make it more explicit - this can be extrapolated as well from the given examples

File: sh/tests/array-selector_test.sh (hunk 3 above)
Comment (nprimo) [reply sh/tests/array-selector_test.sh:1]: The empty solution issue was related to the `exit` in the previous solution version - now it should be fixed

**Review by MSilva95 — CHANGES_REQUESTED** [review]

**Review by MSilva95 — APPROVED** [review]
LGTM!
